File: CS/Form1.cs
using System;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;
using DevExpress.XtraRichEdit.API.Native;
using DevExpress.XtraRichEdit.Utils;
using DevExpress.Office.Utils;

namespace RichEditLeftPadding {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();

            AdjustSimpleViewPadding();
            AdjustDraftViewPadding();

            richEditControl1.ActiveViewType = RichEditViewType.Simple;

            richEditControl1.Text = "Test";
        }

        private void richEditControl1_DocumentLoaded(object sender, EventArgs e) {
            //AdjustParagraphIndent();
            AdjustMargins();
        }

        private void barButtonItem1_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            AdjustParagraphIndent();
        }

        private void AdjustSimpleViewPadding() {
            richEditControl1.Views.SimpleView.Padding = new Padding(0);
        }

        private void AdjustDraftViewPadding() {
            richEditControl1.Views.DraftView.Padding = new Padding(0);
        }

        private void AdjustParagraphIndent() {
            richEditControl1.Document.Paragraphs[0].LeftIndent = Units.InchesToDocumentsF(0.5f);
            richEditControl1.Document.Paragraphs[0].FirstLineIndentType = ParagraphFirstLineIndent.Indented;
            richEditControl1.Document.Paragraphs[0].FirstLineIndent = Units.InchesToDocumentsF(0.5f);
        }

        private void AdjustMargins() {
            richEditControl1.Document.Sections[0].Margins.Left = Units.InchesToDocumentsF(2f);
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit
Imports DevExpress.XtraRichEdit.API.Native
Imports DevExpress.Office.Utils

Namespace RichEditLeftPadding

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            AdjustSimpleViewPadding()
            AdjustDraftViewPadding()
            richEditControl1.ActiveViewType = RichEditViewType.Simple
            richEditControl1.Text = "Test"
        End Sub

        Private Sub richEditControl1_DocumentLoaded(ByVal sender As Object, ByVal e As EventArgs)
            'AdjustParagraphIndent();
            AdjustMargins()
        End Sub

        Private Sub barButtonItem1_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            AdjustParagraphIndent()
        End Sub

        Private Sub AdjustSimpleViewPadding()
            richEditControl1.Views.SimpleView.Padding = New Padding(0)
        End Sub

        Private Sub AdjustDraftViewPadding()
            richEditControl1.Views.DraftView.Padding = New Padding(0)
        End Sub

        Private Sub AdjustParagraphIndent()
            richEditControl1.Document.Paragraphs(0).LeftIndent = Units.InchesToDocumentsF(0.5F)
            richEditControl1.Document.Paragraphs(0).FirstLineIndentType = ParagraphFirstLineIndent.Indented
            richEditControl1.Document.Paragraphs(0).FirstLineIndent = Units.InchesToDocumentsF(0.5F)
        End Sub

        Private Sub AdjustMargins()
            richEditControl1.Document.Sections(0).Margins.Left = Units.InchesToDocumentsF(2F)
        End Sub
    End Class
End Namespace
